src/timeaxis/timegrid.jl

```diff
@@ -287,6 +299,45 @@ Base.resize!(tg::TimeGrid{T,P,:infinite}, n::Int) where {T,P} = tg
 
 Base.diff(tg::TimeGrid{T,P,:finite}) where {T,P} = fill(tg.p, tg.n - 1)
 
+###############################################################################
+#  Intersect
+###############################################################################
+function Base.intersect(tg1::TimeGrid, tgs...)
+    for tg in tgs
+        tg1 = intersect(tg1, tg)
+    end
+    return tg1
+end
+# TODO 
+# Consider checking if they are equal
+function intersect(tg_inf::TimeGrid{T,P,:infinite}, 
+                   tg_fin::TimeGrid{T,P,:finite})::TimeGrid{T,P,:finite} where {T,P}
+    @assert tg_fin.p % tg_inf.p
+    if tg_fin[end] < tg_inf.o
+        return TimeGrid(tg_fin.o, tg_fin.p, 0)
+    end
+    new_o = max(tg_fin.o, tg_inf.o)
+    new_n = tg_inf.o > tg_fin.o ? findfirst(isequal(tg_fin[end]), tg_inf) : 
+                                  findfirst(isequal(tg_fin[end]), tg_fin)
+    return TimeGrid(new_o, tg_fin.p, new_n)
+end
+intersect(tg_fin::TimeGrid{T,P,:finite}, tg_inf::TimeGrid{T,P,:infinite}) where {T,P} = intersect(tg_inf, tg_fin)
+function intersect(tg_inf1::TimeGrid{T,P,:infinite}, tg_inf2::TimeGrid{T,P,:infinite}) where {T,P}
+    @assert tg_inf1.p == tg_inf2.p
+    new_o = max(tg_inf1.o, tg_inf2.o)
+    return TimeGrid(new_o, tg_inf1.p)
+end
+function intersect(tg_fin1::TimeGrid{T,P,:finite}, tg_fin2::TimeGrid{T,P,:finite}) where {T,P}
+    @assert tg_fin1.p == tg_fin2.p
+    if (tg_fin2[end] < tg_fin1.o) || (tg_fin1[end] < tg_fin2.o)
+        return TimeGrid(tg_fin1.o, tg_fin1.p, 0)
```

Comment:
There is a constructor that can help for this case
```suggestion
        return TimeGrid(tg_fin1, n = 0)
```